### PR TITLE
Readonlyaccounts

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
prepare for further abstraction of AccountSharedData
#### Summary of Changes
lamports -> lamports()
Fixes #

### CODE REVIEW

File: runtime/src/message_processor.rs [hunk 1]
```diff
@@ -125,13 +125,13 @@ impl PreAccount {
...
         if *program_id != pre.owner // line coverage used to get branch coverage
-         && pre.lamports > post.lamports
+         && pre.lamports() > post.lamports
         {
             return Err(InstructionError::ExternalAccountLamportSpend);
         }
 
         // The balance of read-only and executable accounts may not change
-        let lamports_changed = pre.lamports != post.lamports;
+        let lamports_changed = pre.lamports() != post.lamports;
```
Comment (brooksprumo) [inline runtime/src/message_processor.rs:134]: Does `post.lamports` need to be `post.lamports()`?

File: runtime/src/message_processor.rs [hunk 2]
```diff
@@ -125,13 +125,13 @@ impl PreAccount {
 
         // An account not assigned to the program cannot have its balance decrease.
         if *program_id != pre.owner // line coverage used to get branch coverage
-         && pre.lamports > post.lamports
+         && pre.lamports() > post.lamports
```
Comment (brooksprumo) [inline runtime/src/message_processor.rs:128]: Does `post.lamports` need to be `post.lamports()`?

File: runtime/src/rent_collector.rs [hunk 3]
```diff
@@ -158,7 +158,7 @@ mod tests {
         assert_ne!(existing_account.rent_epoch, old_epoch);
 
         // newly created account should be collected for less rent; thus more remaining balance
-        assert!(created_account.lamports > existing_account.lamports);
+        assert!(created_account.lamports() > existing_account.lamports);
```
Comment (brooksprumo) [inline runtime/src/rent_collector.rs:161]: Does `existing_account.lamports` need to be `existing_account.lamports()`?

**Review by brooksprumo — COMMENTED** [review]
I only looked at the changes via this PR, not the files in their entirety, so I may've missed other uses of bare `.lamports`.

File: runtime/src/message_processor.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/message_processor.rs:134]: maybe, will pick these up on a future pr when I do another batch of stragglers.

**Review by brooksprumo — APPROVED** [review]

**Pull request merged**